src/core/analysis/mcmc/Mcmcmc.cpp

```diff
@@ -530,6 +612,14 @@ void Mcmcmc::nextCycle(bool advanceCycle)
         }
     }
     
+    for(int i=0;i<num_chains;i++)
+    {
+        int heat_rank = heat_ranks[i];
+        if (chain_prev_boundary[i] == boundary::coldest)
+            heat_visitors[heat_rank].first++;
+        if (chain_prev_boundary[i] == boundary::hottest)
+            heat_visitors[heat_rank].second++;
+    }
```

Comment:
Not entirely sure if heat_ranks needs to be synchronized (explicitly with a specific synchronization function) across processes, like the chain_heats, when MPI is used (seems unnecessary to me, but I never fully grasped the reason that the heats need to be explicitly synchronized...).

Reply:
Good point.  Relying on heat_ranks[] here might not be correct, since the heat_ranks[] might be out of date with respect to the chain_heats[].

It is problematic that the same information is stored in multiple places -- they can get out of sync.  I started doing additional refactoring to 
* compute chain_heat(i) from heat_temps[] and heat_ranks[]
* synchronize heat_temps[] and heat_ranks[]
However, I decided to avoid doing such invasive changes in the first round.  The code could really use a second round of refactoring.

Reply:
OK, this should be fixed now.